### PR TITLE
Add Novita provider

### PR DESCRIPTION
**Pull request by hanouticelina**

### CODE REVIEW

File: tests/test_inference_client.py [hunk 1]
```diff
@@ -326,7 +330,6 @@ def test_chat_completion_with_stream(self, client: InferenceClient):
             assert len(item.choices) == 1
             assert item.choices[0].finish_reason is None
             assert item.choices[0].index == 0
-            assert item.choices[0].delta.role == "assistant" or item.choices[0].delta.content is not None
```
Comment (hanouticelina) [inline tests/test_inference_client.py:329]: removed this line as we have more corner cases with novita (in addition to fireworks and hyperbolic). imo it's okay to remove this check

**Review by Wauplin — APPROVED** [review]
Quick and clean, thanks!

**Review by Wauplin — APPROVED** [review]
Thanks!
